### PR TITLE
Add syntax highlighting support for BPXPRM and IEASYS

### PR DESCRIPTION
**Pull request by ydjainopensource**
Signed-off-by: Yash Jain <ydjainopensource@gmail.com>

### CODE REVIEW

File: webClient/src/app/editor/code-editor/monaco/monaco.config.ts [hunk 1]
```diff
@@ -31,6 +31,205 @@ const JCL_LANG = {
   mimetypes: ['application/jcl']
 };
 
+const IEASYS_LANG = {
+  // TODO
+  id: 'IEASYS',
+  extensions: [],
+  filenamePatterns: ['IEASYS\\d\\d$'],
+  aliases: ['IEASYS'],
+  mimetypes: ['application/']
```
Comment (ydjainopensource) [inline webClient/src/app/editor/code-editor/monaco/monaco.config.ts]: @1000TurquoisePogs what should be correct mimetype here?

File: webClient/src/app/editor/code-editor/monaco/monaco.config.ts [hunk 2]
```diff
@@ -31,6 +31,205 @@ const JCL_LANG = {
...
+};
+
+const BPXPRM_LANG = {
+  // TODO
+  id: 'BPXPRM',
+  extensions: [],
+  filenamePatterns: ['BPXPRM\\d\\d$'],
+  aliases: ['BPXPRM'],
+  mimetypes: ['application/']
```
Comment (ydjainopensource) [inline webClient/src/app/editor/code-editor/monaco/monaco.config.ts]: and here

File: webClient/src/app/editor/code-editor/monaco/monaco.config.ts (hunk 1 above)
Comment (1000TurquoisePogs) [reply webClient/src/app/editor/code-editor/monaco/monaco.config.ts]: I guess application/ieasys, but it doesn't really exist so we are making this new. We don't sense by mimetype currently, instead by filename pattern.

Comment (ydjainopensource) [reply webClient/src/app/editor/code-editor/monaco/monaco.config.ts]: Okay I will make the change and repush

**Comment by ydjainopensource** [comment]
@1000TurquoisePogs changes made. PTAL.

**Comment by 1000TurquoisePogs** [comment]
Hi @ydjainopensource do you know how I can test this? This is the function that determines which language to choose for highlighting: https://github.com/zowe/zlux-editor/blob/master/webClient/src/app/shared/editor-control/editor-control.service.ts#L842

When we compare JCL vs IEASYS, I see a different filename pattern, and no extensions, so I am not sure if it detects IEASYS properly. Do you have a good test?


const JCL_LANG = {
  id: 'jcl',
  extensions: ['.jcl', '.cntl'],
  filenamePatterns: ['\\.jcl\\.','\\.jcl','\\.cntl','\\.cntl\\.'],
  aliases: ['JCL', 'jcl'],
  mimetypes: ['application/jcl']
};

const IEASYS_LANG = {  
  id: 'ieasys',
  extensions: [],
  filenamePatterns: ['IEASYS\\d\\d$'],
  aliases: ['IEASYS'],
  mimetypes: ['application/ieasys']
};

**Pull request merged**